includes/tracking-events.php

```diff
@@ -1,120 +1,87 @@
             );
-            $this->single_item_tracked = true;
-            
-        } elseif (
-            !$this->single_item_tracked &&
-            function_exists('is_checkout_pay_page') &&
-            is_checkout_pay_page()
-        ) {
+        } elseif (function_exists('is_checkout_pay_page') && is_checkout_pay_page()) {
             $this->put_event_in_queue(
                 "window.metrilo.customEvent('checkout_payment');"
             );
-            $this->single_item_tracked = true;
-        } elseif (!$this->single_item_tracked && is_checkout()) {
+        } elseif (is_checkout()) {
             $this->put_event_in_queue(
                 "window.metrilo.checkout();"
             );
-            $this->single_item_tracked = true;
         }
     }
     //
     //  ** GENERIC WordPress tracking - doesn't require WooCommerce in order to work **//
     //
     //  if visitor is viewing homepage or any text page
-    if (!$this->single_item_tracked && is_front_page()) {
+    if (is_front_page()) {
+        $url  = get_permalink();
+        $name = json_encode(['name' => get_the_title()]);
         $this->put_event_in_queue(
-            "window.metrilo.viewPage('"
-            . get_permalink()
-            . "', "
-            . json_encode(['name' => get_the_title()])
-            . ");"
+            "window.metrilo.viewPage('$url', $name);"
         );
-        $this->single_item_tracked = true;
-    } elseif (!$this->single_item_tracked && is_page()) {
+    } elseif (is_page() && !is_cart() && !is_checkout()) {
+        $url  = get_permalink();
+        $name = json_encode(['name' => get_the_title()]);
         $this->put_event_in_queue(
-            "window.metrilo.viewPage('"
-            . get_permalink()
-            . "', "
-            . json_encode(['name' => get_the_title()])
-            . ");"
+            "window.metrilo.viewPage('$url', $name);"
         );
-        $this->single_item_tracked = true;
     }
 
     //  if visitor is viewing post
-    if (!$this->single_item_tracked && is_single()) {
-        $post_id = get_the_id();
+    if (is_single() && !is_product()) {
+        $post_id      = get_the_id();
+        $name_and_url = json_encode(['name' => get_the_title(), 'url' => get_permalink($post_id)]);
         $this->put_event_in_queue(
-            "window.metrilo.viewArticle('"
-            . $post_id
-            . "', "
-            . json_encode(['name' => get_the_title(), 'url' => get_permalink($post_id)])
-            . ");"
+            "window.metrilo.viewArticle('$post_id', $name_and_url);"
         );
-        $this->single_item_tracked = true;
     }
     
-    //  check if there are events in the queue to be sent to Metrilo
     if ($this->identify_call_data !== false) {
         $this->render_identify();
     }
     
-    if (count($this->events_queue) > 0) {
+    if (!empty($this->events_queue)) {
         $this->render_events();
-    }
\ No newline at end of file
+    }
```

Comment:
Do you need this check? 

`$this->events_queue` is an array and in `render_events()` you iterate over it and send Metrilo events. If the array is empty, foreach won't do anything, so I don't see why you still check if it's empty.